### PR TITLE
Fixed Jester respawning as Jester loop for winstates 1, 2, 5

### PR DESCRIPTION
**Pull request by Histalek**
This should fix the annoying bug where the jester would respawn as jester in winstates 1, 2, 5. This happened when no one or only players in the same team as the player who killed the jester died. In that case there was no 'open' role for the jester to respawn as. 

this definetly needs improvement though.
(the first of the 2 for-loops in the winstate functions just should not exist as such... )

more testing can't be negative either ^^

### CODE REVIEW

File: lua/terrortown/entities/roles/jester/winstates.lua [hunk 1]
```diff
@@ -198,18 +196,17 @@ function JesterWinstateFive(ply, killer)
 	local reviveRoleCandidates = GetSelectableRoles()
 	local reviveRoles = {}
 
+	local innoTbl = roles.GetByName("Innocent")
```
Comment (Alf21) [inline lua/terrortown/entities/roles/jester/winstates.lua]: You could just use `INNOCENT` or `TRAITOR`

File: lua/terrortown/entities/roles/jester/winstates.lua [hunk 2]
```diff
@@ -94,18 +94,23 @@ end
 --Player spawns within three seconds with a random opposite role of the killer
 function JesterWinstateOne(ply, killer)
 	local rd = killer:GetSubRoleData()
-	local avoidedRoles = {}
-
-	for _, v in pairs(roles.GetList()) do
-		if v.defaultTeam == rd.defaultTeam then
-			avoidedRoles[v] = true
+	local reviveRoleCandidates = GetSelectableRoles()
```
Comment (Alf21) [inline lua/terrortown/entities/roles/jester/winstates.lua]: table.Copy around GetSelectableRoles missing, you modify the original selection table :D (ouch ^^)

**Review by Alf21 — CHANGES_REQUESTED** [review]

File: lua/terrortown/entities/roles/jester/winstates.lua [hunk 3]
```diff
@@ -94,18 +94,23 @@ end
...
+	local reviveRoles = {}
+
+	-- make sure innocent and traitor are revive candidate roles
+	reviveRoleCandidates[INNOCENT] = reviveRoleCandidates[INNOCENT] or 1
+	reviveRoleCandidates[TRAITOR] = reviveRoleCandidates[TRAITOR] or 1
+	--remove jester from the revive candidate roles
+	reviveRoleCandidates[JESTER] = nil
+
+	for k, _ in pairs(reviveRoleCandidates) do
```
Comment (Alf21) [inline lua/terrortown/entities/roles/jester/winstates.lua]: you could remove the comma and the underscore :)

Comment (Histalek) [reply lua/terrortown/entities/roles/jester/winstates.lua]: just out of curiosity: does this matter performance-wise?

File: lua/terrortown/entities/roles/jester/winstates.lua (hunk 2 above)
Comment (Histalek) [reply lua/terrortown/entities/roles/jester/winstates.lua]: oof, you're right. Fixed.

File: lua/terrortown/entities/roles/jester/winstates.lua (hunk 3 above)
Comment (TimGoll) [reply lua/terrortown/entities/roles/jester/winstates.lua]: Since `_` is a global variable, this can be a bit slower as far as I know. But Alfs notice was mostly for readability

File: lua/terrortown/entities/roles/jester/winstates.lua [hunk 4]
```diff
@@ -94,18 +94,23 @@ end
 --Player spawns within three seconds with a random opposite role of the killer
 function JesterWinstateOne(ply, killer)
 	local rd = killer:GetSubRoleData()
-	local avoidedRoles = {}
-
-	for _, v in pairs(roles.GetList()) do
-		if v.defaultTeam == rd.defaultTeam then
-			avoidedRoles[v] = true
+	local reviveRoleCandidates = Table.Copy(GetSelectableRoles())
```
Comment (Alf21) [inline lua/terrortown/entities/roles/jester/winstates.lua]: `table`, not `Table` ^^

**Review by Alf21 — CHANGES_REQUESTED** [review]

**Comment by Alf21** [comment]
Nice, ty!

**Comment by Histalek** [comment]
Glad to help :) took me long enough =D
